src/graph.jl

```diff
@@ -68,6 +68,72 @@ function rook_intersection(intersection)::Bool
 end
 
 
+function adjacency_exists(node₁_polygons::Array{LibGEOS.Polygon, 1},
+                          node₂_polygons::Array{LibGEOS.Polygon, 1},
+                          adjacency_fn::Function)::Bool
+    """ Returns true if there exists an adjacency between any of the polygons
+        in node 1 and any of the polygons in node 2. The user provides
+        an adjacency_fn (such as queen_intersection or rook_intersection) to
+        evaluate the output of the LibGEOS.intersection() function.
+    """
+    for p₁ in node₁_polygons
+        for p₂ in node₂_polygons
+            intersection = LibGEOS.intersection(p₁, p₂)
+            if adjacency_fn(intersection)
+                return true
+            end
+        end
+    end
+    return false
+end
+
+
+function simple_graph_from_shapes(polygons::Array{Array{LibGEOS.Polygon, 1}, 1},
```

Comment:
Not sure about this name. By "simple graph," do you mean that in the formal graph-theoretical sense (a graph with no self-loops)? In that case, is there any reason why a _non_-simple graph would arise? Also, I would replace `_from_shapes` with `_from_polygons` just to keep the nomenclature consistent.

Reply:
co-sign!

Reply:
you got it! 👍